gis/openstreetmap-carto/Portfile

```diff
@@ -0,0 +1,203 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           active_variants 1.1
+PortGroup           github 1.0
+PortGroup           python 1.0
+
+github.setup        gravitystorm openstreetmap-carto 5.3.1 v
+github.tarball_from archive
+revision            0
+
+categories          gis
+platforms           darwin
+maintainers         {@frankdean fdsd.co.uk:frank.dean} openmaintainer
+
+# license CC0-1.0 (http://creativecommons.org/publicdomain/zero/1.0/)
+license             public-domain
+
+description         A general-purpose OpenStreetMap mapnik style, in CartoCSS
+
+long_description    These are the CartoCSS map stylesheets for the Standard \
+                    map layer on OpenStreetMap.org.
+
+checksums           rmd160  f3effe425ea8c0d54f5437f22fe0181bb1005ea3 \
+                    sha256  be63d4fc06f1db447f1a249515f9fc83449f7f3027469d1cc051150ded3f85e9 \
+                    size    2719803
+
+python.default_version 39
+
+depends_lib-append  port:py${python.version}-psycopg2 \
+                    port:py${python.version}-requests \
+                    port:py${python.version}-yaml
+
+# Database variants (from the GDAL port)
+set postgresql_suffixes {12 13}
+
+set portsgresql_variants {}
+set postgresql_default_variant "if {"
+
+foreach s ${postgresql_suffixes} {
+    lappend portsgresql_variants postgresql${s}
+    set postgresql_default_variant "${postgresql_default_variant}!\[variant_isset postgresql${s}] && "
+}
+
+set postgresql_default_variant [string range ${postgresql_default_variant} 0 end-4]
+set postgresql_default_variant "${postgresql_default_variant}} {default_variants +postgresql${s}}"
+eval $postgresql_default_variant
+
+foreach s ${postgresql_suffixes} {
+    set p postgresql${s}
+    set v [string index ${s} 0].[string index ${s} 1]
+    set i [lsearch -exact ${portsgresql_variants} ${p}]
+    set c [lreplace ${portsgresql_variants} ${i} ${i}]
+    variant ${p} description "Enable PostgreSQL ${v} support" conflicts {*}${c} "
+        depends_lib-append      port:${p}
+    "
+}
+
+set datadir         ${prefix}/var/lib/${name}
+set configdir       ${prefix}/etc/${name}
+
+use_configure       no
+
+# The `scripts-get-external-data-py.diff` patch has been merged
+# into the upstream master at
+# https://github.com/gravitystorm/openstreetmap-carto/commit/4ec2dc9391c411e124c78b3ba1aad9173fea20cb
+# so it should be possible to remove it after the next release (> v5.3.1).
+patchfiles          style-fonts-mss.diff \
+                    scripts-get-external-data-py.diff
+
+variant postgresql12 conflicts postgresql13 description {Use with PostgreSQL 12} {
+    require_active_variants py${python.version}-psycopg2 postgresql12
+}
+
+variant postgresql13 conflicts postgresql12 description {Use with PostgreSQL 13} {
+    require_active_variants py${python.version}-psycopg2 postgresql13
+}
+
```

Comment:
add the lines below to set the default variant if the user doesn't pick one:
```
if {![variant_isset postgresql12] && ![variant_isset postgresql13]} {
    default_variants +postgresql13
}
```

Reply:
Ok